src/dotty/tools/dotc/core/Types.scala

```diff
@@ -869,7 +869,13 @@ object Types {
         case SkolemType(tp) =>
           tp.lookupRefined(name)
         case pre: WildcardType =>
-          WildcardType
+          pre.optBounds match {
+            case TypeBounds(lo, hi) if name.isTypeName =>
+              // for term names, forming a Termref with just the name risks losing overloaded instance
+              WildcardType(TypeBounds(NamedType(lo, name), NamedType(hi, name)))
```

Comment:
So, does this mean that we consider `(_ >: Foo <: Bar)#T` to be equivalent to `_ >: Foo#T <: Bar#T` ? If so, isn't this wrong in general? For example, if `T` is contravariant we may have `Foo <: Bar` but `Foo#T >: Bar#T` so the bounds `_ >: Foo#T <: Bar#T` would not be satisfiable.